xrootd/xrdfs/file.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package xrdfs
+
+import (
+	"go-hep.org/x/hep/xrootd/internal/xrdenc"
+)
+
+// File implements access to a content and meta information of file over XRootD.
+type File interface {
+	Compression() *FileCompression // Compression returns the compression info.
+	Info() *EntryStat              // Info returns the stat info.
+	Handle() FileHandle            // Handle returns the file handle.
+}
+
+// FileHandle is the file handle, which should be treated as opaque data.
+type FileHandle [4]byte
+
+// FileCompression holds the compression parameters such as the page size and the type of compression.
+type FileCompression struct {
+	PageSize int32
```

Comment:
it would be interesting to document in what unit (kB, MB, ...) this page size is.

Reply:
I was assuming that it's in bytes but it's unclear from the documentation (`cpsize` part): http://xrootd.org/doc/dev45/XRdv310.htm#_Toc464248823 :(

Same goes for the `cptype` - it's unclear from the documentation whether it's name or not.

I have tried to call `open` with `kxr_compress` option on `gzip`-ed file some time ago but haven't got any success with that.

Reply:
feel free to send an issue asking for some clarifications of the specs :)

Reply:
I've filled an issue: https://github.com/xrootd/xrootd/issues/724.

Reply:
thanks.